articles/modules/ROOT/pages/migrating-explicit-lucene-indexes-to-native-schema-indexes.adoc

```diff
@@ -0,0 +1,102 @@
+
+Thus any index migration would have to be done on an index by index basis and might involve some level of guessing.   This means you having to look at the API code and assess which node or relationship properties are being indexes, and accordingly implement a complementing schema index on those properties and/or look at the cypher query and come up with supporting indexes to optimize its execution.
+
+Overall at a high level, here is the recommended approach:
+
+1) Get a listing of all your indexes on you existing version
+
+*  Example:   in 3.x you can use **"call db.index.explicit.list();"** which will show the index name as well as its type, which can be either “exact” or “full-text”, as well as if it is a node or relationship index, which will help you to determine what type of schema index to construct. 
+ 
+2) Next look at your cypher and/or Java API code and convert them accordingly to 4.x+ format.   
+
+* For example, convert "start" statements to equivalent Match statement.  Example: 
+**  **START n=node:myExplicitIndexYear("myid:1234567") RETURN n;**
+** The above query would be changed to the following:
+**  **MATCH (n:Person{myid:1234567}) RETURN n;**
+
+ 
+
+3) Implement the equivalent schema index that you would need to support and optimize the execution of the associated/converted queries from step 2. 
+
+* Example:   **CREATE INDEX index_name FOR (n:Person) ON (n.myid);**
+* More specifically:
+** Inspect all cypher queries for what they actually write to that index and derive either a create index (for integer, strings, dates, geospatial, etc) and/or call db.index.fulltext.createIndexForNodes (FTS) statements.
+** Inspect all reads to that index and adopt the statements to use the new index (this would mean to enable query logging to capture generated cypher queries and its performance attributes such as execution time vs io vs memory, vs cpu to flag longest running queries to troubleshoot individually one at a time - and more than likely your query patterns are similar to each other, hence, once one query is fixed, it will similarly help other similar queries.).
+ 
+4) Once finished with all the explicit indexes, then shut down the database and get rid of associated index subfolder.
+
+please note, if it happens to that you convert a legacy index to an equivalent schema index and your search query using the legacy indexes are returning an unexpected number of rows (compared to schema indexes), then this either means that the schema index is either incorrectly defined (on wrong label:properties) or the legacy index itself is not up to date (remember legacy indexes must be manually kept up to date, hence as an example, without the appropriate error handling in your Java API code, it is entirely possible that a transaction would not have correctly retried and applied a transaction when encountering a rollback due to any reason, such as server restarts, etc).   Here is a 3.x example demonstrating the case in point:
```

Comment:
Suggesting slight rephrasing of this paragraph as:

"Please note, if it happens that having converted a legacy index to an equivalent schema index, the search query using the legacy indexes are returning an unexpected number of rows (compared to schema indexes), then this likely indicates that the schema index may have been incorrectly defined (on wrong label:properties), or, the legacy index is not up to date (remember legacy indexes must be manually kept up to date, hence as an example, without the appropriate error handling in your Java API code, it is entirely possible that a transaction would not have been correctly retried and applied commits when encountering a rollback due to any reason, such as server restarts, etc). Here is a 3.x example demonstrating such case:"